pkg/pool-utils/contracts/rates/PriceRateCache.sol

```diff
@@ -23,55 +23,71 @@ import "@balancer-labs/v2-solidity-utils/contracts/helpers/WordCodec.sol";
  * useful for slow changing rates, such as those that arise from interest-bearing tokens (e.g. waDAI into DAI).
  *
  * The cache data is packed into a single bytes32 value with the following structure:
- * [   expires   | duration | price rate value ]
- * [   uint64    |  uint64  |      uint128     ]
- * [ MSB                                   LSB ]
+ * [ 32 bits |  32 bits  |   96 bits  |    96 bits    ]
+ * [ expires | duration  | last rate  | current rate  ]
```

Comment:
```suggestion
 * [ expires | duration  | old rate  | current rate  ]
```

I'd rename this to 'old', and not make any assumptions here about how we're going to use it. We just have a function to update the current rate, and another to set the old rate to the current rate.

Reply:
Essentially, one rate that is updated according to the cache expiry logic, and another that is updated according to arbitrary business logic.